src/common/mfu_daos.c

```diff
@@ -331,6 +332,8 @@ static int daos_set_api_cont_type(
 
     if (cont_type == DAOS_PROP_CO_LAYOUT_POSIX) {
         mfu_file->type = DFS;
+    } else if (cont_type == DAOS_PROP_CO_LAYOUT_HDF5) {
+        mfu_file->type = HDF5;
```

Comment:
@dsikich Just a note that we'll need to incorporate DAOS_PROP_CO_LAYOUT_HDF5 into the dest logic rework. I don't expect there to be any "conflicts", but some of the logic might need some adjustments.

Reply:
@daltonbohning I think that h5repack should handle creating the destination. I am about to test some of this but this should work once  
DAOS-6574 duns: some bug fixes and features to DUNS is merged into DAOS.

Reply:
@daltonbohning after testing, h5repack does not accept daos://$pool/ without a generated UUID, so I think we will just need to generate a UUID for the destination in this case like we do the other tools.

Reply:
@daltonbohning but we do not need to create the container

Reply:
@dsikich I imagine that's probably because duns_resolve_path currently fails for daos://pool.
I wonder if Mohamad has plans to support daos://pool after his DAOS PR? https://github.com/daos-stack/daos/pull/4798#event-4414016922
But I guess worst case, we can just call duns_resolve_path, check for missing container, generate uuid if needed, rebuild the daos:// path, and pass to h5repack

Reply:
@daltonbohning that is the PR I was using to run the h5repack, so daos://$pool/ will not work for h5repack even using it I guess. So we will still need to generate the cont UUID to pass to h5repack, and that handles container creation.

Reply:
@dsikich As a reference, I have similar logic for doing this in my PR:
https://github.com/hpc/mpifileutils/pull/468/commits/2cfc511d5c1bd185a2e415d826954cc1ab3991e9
(scroll to line ~420)

For now, I would say to just do what you need to and I can handle any conflicts on my end, since my PR doesn't contain "necessary" changes.